apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/TestMessageHandler.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.ClientTesting;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Storage.V1.Tests;
+
+internal class TestMessageHandler : ReplayingMessageHandler
+{
+    private readonly List<int> _expectedTime;
+
+    private DateTime _lastSeenTime;
+    private int _numOfSendAsyncCalls;
+
+    public TestMessageHandler(string header, List<int> expectedTime) : base(header)
+    {
+        _expectedTime = expectedTime;
+        _numOfSendAsyncCalls = 0;
+    }
+
+    public void SetLastSeenTime()
+    {
+        _lastSeenTime = DateTime.UtcNow;
+    }
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        var currTime = DateTime.UtcNow;
+        double actualBackoff = (currTime - _lastSeenTime).TotalSeconds;
+        Assert.Contains(_expectedTime[_numOfSendAsyncCalls], new List<double>() { Math.Floor(actualBackoff), Math.Ceiling(actualBackoff) });
```

Comment:
I suspect we can be a bit finer-grained than this. You can use `Assert.Equals(double, double, double)` to provide a tolerance. (250ms should be enough, I'd expect.) We can eventually replace this with a fake clock anyway, of course.